bin/mirror-to-s3.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -e
+
+./bin/mirror.sh 
+
+aws s3 sync ./_site ${BUCKET_PATH} \
+    --acl public-read \
+    --cache-control "max-age=315360000, public, immutable"  \
```

Comment:
This header should only be applied to files with names that include the content hash. It looks like this will be applied to all files.

Reply:
Nice catch. I believe we can use something like

`aws s3 sync ./site $(jq .paths[] _site/static/staticfiles.json | xargs -I {} echo "--exclude {} \\") --acl public-read`  (not tested, fine tuning maybe needed)

to exclude files with content hashes from first sync and then run the opposite operation to include only those files with the proper `cache-control` headers.

I suggest that we first sync files with content-hashes and then the rest of the files to optimize the case where `index.html` is fresh and refers to a new `css`. With this approach the new css will be on S3 before being ever referenced.

Reply:
I like that! Very smart to use `jq` and the JSON manifest to do this. The other approach I could think of was to have the `collectstatic` operation output to a separate directory fully separate from the rest of the files and to then do separate aws operations on them. Or even just do the s3 sync on the static files directory, then `rm -rf static`, then `s3 sync` the rest. Also good catch on the order of the sync 👍